libsolidity/experimental/codegen/IRGeneratorForStatements.cpp

```diff
@@ -185,7 +185,7 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
 	m_code << "leave\n";
 }
 
-experimental::Type IRGeneratorForStatements::type(ASTNode const& _node) const
+experimental::Type IRGeneratorForStatements::typeAnnotation(ASTNode const& _node) const
```

Comment:
This is a deterioration IMHO - and in general it's not helpful to do whitespace changes or renamings in a shared base branch with parallel work.

Reply:
This and the past comments are apparently outdated, but we really should not push style fixup commits to this. This is not worthwhile and makes collaboration on this near impossible (I'm by the way also not a fan of doing this on *any* PR actually, but here in particular) - and we *did* say that we work with *PR*s against this as base branch.

Reply:
IMO better to fix these things now and keep them out of later PRs. I mean, I know this is a prototype and readability was not a priority here, but really this code could often be much easier to understand with really simple changes, like better order, more linebreaks, less ambiguous names. So a lot of it is coming from my frustrations as I was reading it. It really combines in a bad way to the fact that this is a new, completely unfamiliar thing that requires a lot of getting used to and even some retries once you understand more of it. So I just wanted to address some of it in the least obtrusive way.

I thought about submitting these changes as separate PRs, but they seemed inconsequential enough to be better off just squashed into the original branch (assuming there are no objections of course - looks like some were not as uncontroversial as I thought). And it would have caused the same conflicts in a PR as in a fixup.

The collaboration is still minimal at this point, so I don't think it's that bad. Certainly not to the point of making anything impossible. The recent issues with conflicts turned out to be really just the matter of proper use of rebase and that seems resolved now.

Reply:
> This is a deterioration IMHO

Is `type()` with no qualifier really a good name though? It's coming from an annotation so it does not seem quite general enough to deserve such a name. This was quite misleading to me, and initially I often assumed it must be something directly from the type system. Maybe you're just too used to it since you built it, but to me many of these names are really black boxes and I wish they included more context.

But anyway, since you disagree I'll remove it from the base PR and submit separately.

Reply:
Yes, ``type()`` is the best name here. It's also following coding standards established elsewhere as e.g. in the legacy Solidity type checker.

And collaboration is far from a minimal problem in this. I literally cannot reasonably review multiple changes on top of the PR due to unnecessary cosmetics on the base branch.

Reply:
At the moment, the result is pure chaos and I cannot even tell where to start to look at changes that were meant as cosmetic changes, but break design primitives.

Reply:
Honestly I don't really agree with that general view. I mean it's definitely not the top priority, but I also don't agree that spending *some* reasonable effort on keeping things readable is not worthwhile. There's a noticeable overhead to having to chase down things that could otherwise be obvious at the point of use and I was directly affected by that. Some of it is subjective, but tweaking the main pain points as I read though the code was pretty easy and IMO the result is an improvement.

But well, there definitely is some subjectivity to it and - fair enough - if it's not easy to review for you, maybe it was not the best approach. I don't think it's as bad as you think, but I see you don't want to dig into it so I'll change how I'm doing it. I'll remove the changes from the base PR in any case. Honestly, this approach was a bit of an experiment in itself. I was hoping that I can improve things in a clear way without bothering you too much.

> but in fact break design primitives.

Which design primitives are broken?

Reply:
> Which design primitives are broken?

With the change here none - with the ``TypeVariable`` changes more so for example.

The thing about reviewing is also not only about reviewing these cosmetic changes in particular, but this is actively slowing down work on the more substantial changes to the prototype. This mode of work forces everyone who has a PR targetting this branch to permanently watch out for changes for rebasing their respective PRs, s.t. the diffs remain readable and reviewable.

Reply:
It's a trade-off. A lot of these things were at the same time slowing my work on understanding the prototype, and I suspect not only for me, which motivated me to do something about it. And I don't think that retargetting PRs is that much different from constantly rebasing on `develop` or `newAnalysis`. It's all just rebase.

Don't get me wrong - I do see that there's some disconnect here and this mode of work did not work out as smoothly as I expected, so I'm not dismissing what you're saying. I retracted the changes and I'll do it differently. I just don't agree it's completely unworkable and there are some real pain points for me that it was addressing.